inngest/src/main/kotlin/com/inngest/Step.kt

```diff
@@ -102,10 +102,7 @@ class Step(
         val hashedId = state.getHashFromId(id)
 
         try {
-            val stepResult = state.getState(hashedId, type)
-            if (stepResult != null) {
-                return stepResult
-            }
```

Comment:
looks like we inadvertently introduced this bug with https://github.com/inngest/inngest-kt/pull/19

Reply:
i believe it was always present because even the older version check returns `false` when type checking a `null` result, because `null` can never be a specific type:
```kotlin
val stepResult = state.getState<T>(hashedId)
if (stepResult is T) {
```

Reply:
awesome, thanks for verifying